src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/client/prisonapi/PrisonApiMapper.kt

```diff
@@ -0,0 +1,71 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.client.prisonapi
+
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.prisonapi.resource.model.PrisonerInPrisonSummary
+import uk.gov.justice.digital.hmpps.prisonapi.resource.model.SignificantMovement
+import uk.gov.justice.digital.hmpps.prisonapi.resource.model.TransferDetail
+
+@Component
+class PrisonApiMapper {
+  fun toPrisonMovementEvents(
+    prisonerNumber: String,
+    prisonerInPrisonSummary: PrisonerInPrisonSummary,
+  ): PrisonMovementEvents {
+    val prisonMovements = mutableMapOf<Long, List<PrisonMovementEvent>>()
+
+    prisonerInPrisonSummary.prisonPeriod?.forEach {
+      val movementEvents = mutableListOf<PrisonMovementEvent>()
+      movementEvents.addAll(buildArrivalsAndReleases(it.movementDates))
+      movementEvents.addAll(buildTransfers(it.transfers))
+      prisonMovements[it.bookingId!!] = movementEvents
```

Comment:
I will double check with Andy Marke whether `bookingId` will ever be null. If not, I'll create a PR for this spec.

Reply:
If it could be null we might need to rethink our data structure re: creating a map of movements keyed on bookingId 🤔

Reply:
I made the field mandatory in the prison-api: https://github.com/ministryofjustice/prison-api/pull/1988